geth/utils/validation.py

```diff
@@ -94,6 +103,16 @@ class GenesisDataConfig(BaseModel):
     # post-merge, timestamp is used for network transitions
     shanghaiTime: int = 0
     cancunTime: int = 0
+    # blobs
+    blobSchedule: dict[str, Any] = DefaultBlobSchedule().model_dump()
+
+    @model_validator(mode="after")
+    def check_blob_schedule_required(
+        self,
+    ) -> GenesisDataConfig:
+        if self.cancunTime > 0 and self.blobSchedule.get("cancun") is None:
+            raise PyGethValueError("blobSchedule is required when cancunTime is set")
+        return self
 
     model_config = ConfigDict(extra="forbid")
```

Comment:
I'm a bit against these `extra=forbid` additions. I think it may make sense in the `GenesisData` but if there's a new config value that we haven't added yet and it can be passed through to geth, part of me wants to just let it happen in case it doesn't need any validation on our end.

I think our job is to make sure values are formatted well between the user and geth and to raise clear errors if they aren't. But I don't necessarily think we need to do a very rigid, gate-keeping, validation as the middle communication layer between python and geth.

Thoughts on that?

Reply:
Yep, agree. Defaulted to being strict, but allowing new config options through sounds ok.